internal/exec/stages/files/passwd.go

```diff
@@ -66,12 +66,23 @@ func (s *stage) createPasswd(config types.Config) error {
 		}
 
 		s.relabel(deglobbed...)
-		s.relabel(
-			"/etc/.pwd.lock",
-			// for OSTree-based systems
-			"/var/home",
-			"/var/roothome",
-		)
+		s.relabel("/etc/.pwd.lock")
+		for _, user := range config.Passwd.Users {
+			if user.NoCreateHome != nil && *user.NoCreateHome == true {
+				continue
+			}
+			if user.Name == "root" {
+				rootdir_deglobbed, err := s.expandGlobList("/root/*")
+				if err != nil {
+					return err
+				}
+				s.relabel(rootdir_deglobbed...)
+			} else if user.HomeDir != nil && *user.HomeDir != "" {
+				s.relabel(filepath.Join(*user.HomeDir))
+			} else {
+				s.relabel(filepath.Join("/home", user.Name))
```

Comment:
Ideally we could look up the home directory via getpwent.  But at a minimum, let's add `/root` and `/home` to the `distro` package and avoid the hardcoding.

Reply:
Thanks for the suggestion. I changed this to use `userLookup` instead to retrieve the home directory.